daemon/network.go

```diff
@@ -866,20 +866,18 @@ func buildCreateEndpointOptions(c *container.Container, n *libnetwork.Network, e
 		createOptions = append(createOptions, libnetwork.CreateOptionDisableResolution())
 	}
 
-	// configs that are applicable only for the endpoint in the network
-	// to which container was connected to on docker run.
-	// Ideally all these network-specific endpoint configurations must be moved under
-	// container.NetworkSettings.Networks[n.Name()]
-	if nwName == c.HostConfig.NetworkMode.NetworkName() || (nwName == defaultNetName && c.HostConfig.NetworkMode.IsDefault()) {
-		if c.Config.MacAddress != "" {
-			mac, err := net.ParseMAC(c.Config.MacAddress)
-			if err != nil {
-				return nil, err
-			}
-			createOptions = append(createOptions, libnetwork.EndpointOptionGeneric(options.Generic{
-				netlabel.MacAddress: mac,
-			}))
+	if c.Config.MacAddress != "" {
+		if c.HostConfig.NetworkMode.IsHost() || c.HostConfig.NetworkMode.IsNone() || c.HostConfig.NetworkMode.IsContainer() {
+			return nil, fmt.Errorf("you can't specify a mac-address for a container with network-mode %s", c.HostConfig.NetworkMode)
```

Comment:
I realize the daemon logs an extra `FIXME` when this error is returned because it's not a proper `InvalidParameter`.